### ISSUE TITLE
Tags map not cloned on registration

### ISSUE THREAD

**Issue by brunomartinez-lmi** [mathpl/go-tsdmetrics#4]
Affects: DefaultTaggedRegistry and PrefixTaggedRegistry.
When registring a metrics instance, the given map of tags is not cloned, only the reference is kept, which can lead to weird results when the same map instance is used to register several metrics.

At this line, a copy should be done (or in higher level Register() method):
https://github.com/mathpl/go-tsdmetrics/blob/5f32b115145c471ea6b11790ef4d65bcd6f1285d/tagged_registry.go#L171

**Comment by gesanderson** [comment]
@brunomartinez-lmi We don't use github issues. If you want to report a bug, please file a JIRA ticket in the INFRA project and send it to petros.

**Comment by brunomartinez-lmi** [comment]
> @brunomartinez-lmi We don't use github issues. If you want to report a bug, please file a JIRA ticket in the INFRA project and send it to petros.

Hello, this was more for a reminder as we implemented workaround, so not usrgent at all.
So you still want me to create a ticket with low priority ?

**Comment by gesanderson** [comment]
> So you still want me to create a ticket with low priority ?

Yes please, and send the ticket to petros. Otherwise we will never remember that we have this bug.

**Comment by gesanderson** [comment]
Same thing for the other issue you opened
